src/crimson/osd/osd_operations/client_request_common.cc

```diff
@@ -17,19 +17,25 @@ CommonClientRequest::do_recover_missing(
   Ref<PG>& pg, const hobject_t& soid)
 {
   eversion_t ver;
+  assert (pg->is_primary());
   logger().debug("{} check for recovery, {}", __func__, soid);
-  if (!pg->is_unreadable_object(soid, &ver) &&
-      !pg->is_degraded_or_backfilling_object(soid)) {
+  if (pg->is_unreadable_object(soid, &ver)){
+    logger().error("{} {} is unreadable", __func__, soid);
+    //TODO:: wait for unreadable
```

Comment:
Why are we leaving a TODO here?

Reply:
The intent here was to separate the `wait_for_unreadable_object()` and `wait_for_degraded_object()` cases, same as classic does (using`<hobject_t, std::list<OpRequestRef>>` maps).
In crimson there isn't any differentiation between those, we simply "wait for recovery" in both cases.
In second thought, I will leave the current implementation as is and drop this commit since it's out of the scope of this PR.